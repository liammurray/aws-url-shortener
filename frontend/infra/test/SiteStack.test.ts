import { expect as cdkExpect, haveResource, MatchStyle, matchTemplate } from '@aws-cdk/assert'
import { SynthUtils } from '@aws-cdk/assert'

import * as cdk from '@aws-cdk/core'
import SiteStack from '../src/stacks/siteStack'

describe('Build pipeline stack', () => {
  let stack: SiteStack

  const stackProps = {
    env: {
      region: 'us-west-2',
      account: 'account12345',
    },
    siteProps: {
      domain: 'mydomain.com',
      subDomain: 'urls', // urls.mydomain.com
      retainDeployBucket: false,
      certId: 'cert12345',
      codeRoot: '../code/build',
    },
  }
  beforeAll(() => {
    const app = new cdk.App()

    stack = new SiteStack(app, 'test-stack', stackProps)
  })
  xtest('Matches snapshot', () => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot()
  })
  test('Has resources', () => {
    cdkExpect(stack).to(
      haveResource('AWS::Route53::RecordSet', {
        Name: 'urls.mydomain.com.',
      })
    )
  })
  xtest('Match resources', () => {
    cdkExpect(stack).to(
      matchTemplate(
        {
          Resources: {},
        },
        MatchStyle.EXACT
      )
    )
  })
})
