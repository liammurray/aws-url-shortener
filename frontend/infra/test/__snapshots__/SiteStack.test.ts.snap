// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Build pipeline stack Matches snapshot 1`] = `
Object {
  "Outputs": Object {
    "StaticSite93C8948E": Object {
      "Value": "https://urls.mydomain.com",
    },
    "StaticSiteBucket34E5D9AF": Object {
      "Value": Object {
        "Ref": "StaticSiteSiteBucket1A888BC8",
      },
    },
    "StaticSiteDistributionId8C64EF2A": Object {
      "Value": Object {
        "Ref": "StaticSiteSiteDistributionCFDistribution500D676B",
      },
    },
  },
  "Parameters": Object {
    "AssetParameters3f0f86b0b3e14d53e72cc1bcbd00a01ca5773df649bf2be454285c65f0047965ArtifactHash99D04DC5": Object {
      "Description": "Artifact hash for asset \\"3f0f86b0b3e14d53e72cc1bcbd00a01ca5773df649bf2be454285c65f0047965\\"",
      "Type": "String",
    },
    "AssetParameters3f0f86b0b3e14d53e72cc1bcbd00a01ca5773df649bf2be454285c65f0047965S3Bucket227B8CB0": Object {
      "Description": "S3 bucket for asset \\"3f0f86b0b3e14d53e72cc1bcbd00a01ca5773df649bf2be454285c65f0047965\\"",
      "Type": "String",
    },
    "AssetParameters3f0f86b0b3e14d53e72cc1bcbd00a01ca5773df649bf2be454285c65f0047965S3VersionKey7823DA4C": Object {
      "Description": "S3 key for asset version \\"3f0f86b0b3e14d53e72cc1bcbd00a01ca5773df649bf2be454285c65f0047965\\"",
      "Type": "String",
    },
    "AssetParameters85263806834b4abe18b7438876d0e408b131a41c86272285f069bb9fa96666f0ArtifactHash877EFA91": Object {
      "Description": "Artifact hash for asset \\"85263806834b4abe18b7438876d0e408b131a41c86272285f069bb9fa96666f0\\"",
      "Type": "String",
    },
    "AssetParameters85263806834b4abe18b7438876d0e408b131a41c86272285f069bb9fa96666f0S3Bucket88A20322": Object {
      "Description": "S3 bucket for asset \\"85263806834b4abe18b7438876d0e408b131a41c86272285f069bb9fa96666f0\\"",
      "Type": "String",
    },
    "AssetParameters85263806834b4abe18b7438876d0e408b131a41c86272285f069bb9fa96666f0S3VersionKey5726B1E8": Object {
      "Description": "S3 key for asset version \\"85263806834b4abe18b7438876d0e408b131a41c86272285f069bb9fa96666f0\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536": Object {
      "DependsOn": Array [
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters85263806834b4abe18b7438876d0e408b131a41c86272285f069bb9fa96666f0S3Bucket88A20322",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters85263806834b4abe18b7438876d0e408b131a41c86272285f069bb9fa96666f0S3VersionKey5726B1E8",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters85263806834b4abe18b7438876d0e408b131a41c86272285f069bb9fa96666f0S3VersionKey5726B1E8",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
            "Arn",
          ],
        },
        "Runtime": "python3.6",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "AssetParameters3f0f86b0b3e14d53e72cc1bcbd00a01ca5773df649bf2be454285c65f0047965S3Bucket227B8CB0",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "AssetParameters3f0f86b0b3e14d53e72cc1bcbd00a01ca5773df649bf2be454285c65f0047965S3Bucket227B8CB0",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "StaticSiteSiteBucket1A888BC8",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "StaticSiteSiteBucket1A888BC8",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "cloudfront:GetInvalidation",
                "cloudfront:CreateInvalidation",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "Roles": Array [
          Object {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "StaticSiteDeployWithInvalidationCustomResource0B8F0E74": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "DestinationBucketName": Object {
          "Ref": "StaticSiteSiteBucket1A888BC8",
        },
        "DistributionId": Object {
          "Ref": "StaticSiteSiteDistributionCFDistribution500D676B",
        },
        "DistributionPaths": Array [
          "/*",
        ],
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536",
            "Arn",
          ],
        },
        "SourceBucketNames": Array [
          Object {
            "Ref": "AssetParameters3f0f86b0b3e14d53e72cc1bcbd00a01ca5773df649bf2be454285c65f0047965S3Bucket227B8CB0",
          },
        ],
        "SourceObjectKeys": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters3f0f86b0b3e14d53e72cc1bcbd00a01ca5773df649bf2be454285c65f0047965S3VersionKey7823DA4C",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters3f0f86b0b3e14d53e72cc1bcbd00a01ca5773df649bf2be454285c65f0047965S3VersionKey7823DA4C",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        ],
      },
      "Type": "Custom::CDKBucketDeployment",
      "UpdateReplacePolicy": "Delete",
    },
    "StaticSiteSiteAliasRecord5B525E22": Object {
      "Properties": Object {
        "AliasTarget": Object {
          "DNSName": Object {
            "Fn::GetAtt": Array [
              "StaticSiteSiteDistributionCFDistribution500D676B",
              "DomainName",
            ],
          },
          "HostedZoneId": "Z2FDTNDATAQYW2",
        },
        "HostedZoneId": "DUMMY",
        "Name": "urls.mydomain.com.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "StaticSiteSiteBucket1A888BC8": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "BucketName": "urls.mydomain.com",
        "WebsiteConfiguration": Object {
          "IndexDocument": "index.html",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "StaticSiteSiteBucketPolicyD7934718": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "StaticSiteSiteBucket1A888BC8",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Principal": "*",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "StaticSiteSiteBucket1A888BC8",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "StaticSiteSiteDistributionCFDistribution500D676B": Object {
      "Properties": Object {
        "DistributionConfig": Object {
          "Aliases": Array [
            "urls.mydomain.com",
          ],
          "DefaultCacheBehavior": Object {
            "AllowedMethods": Array [
              "GET",
              "HEAD",
            ],
            "CachedMethods": Array [
              "GET",
              "HEAD",
            ],
            "Compress": true,
            "ForwardedValues": Object {
              "Cookies": Object {
                "Forward": "none",
              },
              "QueryString": false,
            },
            "TargetOriginId": "origin1",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Origins": Array [
            Object {
              "DomainName": Object {
                "Fn::GetAtt": Array [
                  "StaticSiteSiteBucket1A888BC8",
                  "RegionalDomainName",
                ],
              },
              "Id": "origin1",
              "S3OriginConfig": Object {},
            },
          ],
          "PriceClass": "PriceClass_100",
          "ViewerCertificate": Object {
            "AcmCertificateArn": "arn:aws:acm:us-east-1:account12345:certificate/cert12345",
            "MinimumProtocolVersion": "TLSv1.1_2016",
            "SslSupportMethod": "sni-only",
          },
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
  },
}
`;
